### PR TITLE
[AMD]  Auto-fallback NSA indexer to page_size=1 when aiter preshuffle gluon kernel is unavailable (Deepseek v3.2)

### PR DESCRIPTION
**Pull request by yctseng0211**
## Motivation
#23562 unconditionally bumped the ROCm NSA indexer to `page_size=64`, `Preshuffle=True`, `KVBlockSize=64`, plus `indexer_k_quant_and_cache(..., preshuffle=True)` and the new `cp_gather_indexer_k_quant_cache` aiter path. All of these require aiter's gluon kernel, which aiter itself gates on:
```python
# aiter/ops/triton/attention/pa_mqa_logits.py
if triton_version >= Version("3.5.0"):
    enable_gluon_pa_mqa_logits = True
else:
    enable_gluon_pa_mqa_logits = enable_aot_gluon_pa_mqa_logits  # AITER_ENABLE_AOT_GLUON_PA_MQA_LOGITS=1
...
if enable_gluon_pa_mqa_logits:
    ...
else:
    assert KVBlockSize == 1
    assert not Preshuffle, "Preshuffle mode is only supported on gluon kernel."
```
The aiter dependency from #23562 (PR ROCm/aiter#2879) is satisfied (the pinned aiter commit `a6bb499` includes it), but the **Triton-version dependency wasn't documented or enforced**, so on the default `rocm700` CI base image (Triton<3.5, no AOT bundle) we hit the legacy aiter branch and crash.
This PR makes the NSA indexer auto-detect at startup whether the preshuffle path is usable, and gracefully falls back to the previous `page_size=1` / `KVBlockSize=1` / `Preshuffle=False` configuration when it isn't, so the rocm700 CI matrix recovers without changing `docker/rocm.Dockerfile` or the base image. On Triton≥3.5 / AOT-enabled environments the new #23562 path is used verbatim.
## Accuracy Tests
rocm7.0
https://github.com/sgl-project/sglang/actions/runs/25836114224
image
rocm7.2
https://github.com/sgl-project/sglang/actions/runs/25837815199
image

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

### CODE REVIEW

**Review by HaiShaw — APPROVED** [review]

**Review by 1am9trash — APPROVED** [review]
LGTM
